src/components/LocalePicker.js

```diff
@@ -17,11 +17,17 @@ export class LocalePicker extends Component {
     const {
       availableLocales,
       classes,
-      locale,
       setLocale,
+      userLanguages,
     } = this.props;
+    let { locale } = this.props;
 
     if (!setLocale || availableLocales.length < 2) return <></>;
+    // If `locale` is not among the available locales, it should be the first available
+    // locale that matches the language preferences from the `userLanguages` store value
+    if (availableLocales.indexOf(locale) < 0) {
+      locale = userLanguages.find(l => availableLocales.indexOf(l) >= 0) ?? availableLocales[0];
```

Comment:
Should we push a default fallback locale into settings? It probably shouldn't happen, but 🤷

Reply:
Well, there's already the `language` value in the store that determines the language of the whole app, which is included in the `userLanguages` list. I'm not sure if an additional `metadataFallbackLocale` really makes sense, wouldn't it be identical to `language` in most usage scenarios?